### PR TITLE
Adding custom repository support

### PR DESCRIPTION
**Pull request by djw8605**
WIP

Adding custom repository support.

- [x] Add support for non-dockerhub auth (required for pulling)
- [x] Parse non-dockerhub repository from container list.

### CODE REVIEW

**Comment by djw8605** [comment]
Ready for review.

File: cvmfs-singularity-sync [hunk 1]
```diff
@@ -253,21 +263,24 @@ def parse_image(image):
...
 
 
 # IMAGE METADATA -------------------------------------------
 # Use Docker Registry API (version 2.0) to get images ids, manifest
 
     # Get an image manifest - has image ids to parse, and will be
     # used later to get Cmd
-    hub = dockerhub.DockerHub(url="https://registry.hub.docker.com")
+    # Prepend "https://" to the registry
+    if not registry.startswith("https://"):
```
Comment (matyasselmeci) [inline cvmfs-singularity-sync]:
```suggestion
    if "://" not in registry:
```
in case it uses a different protocol.

File: dockerhub.py [hunk 2]
```diff
@@ -278,14 +284,15 @@ def _do_request(self, method, address, **kwargs):
...
-                scope = None
                 # Update the auth token with the scope, and try again
                 try:
-                    scope = resp.headers['Www-Authenticate'].split(',')[2]
-                    scope = scope.split('=')[1].replace("\"", "")
+                    # Parse the Www-Authenticate line, looks like:
+                    # Bearer realm="https://git.ligo.org/jwt/auth",service="container_registry",scope="repository:lscsoft/lalsuite/lalsuite-v6.53:pull",error="invalid_token"
+                    reg=re.compile('(\w+)[=] ?"?([\w\:\/\.\-]+)"?')
+                    values = dict(reg.findall(resp.headers['Www-Authenticate']))
                 except:
```
Comment (matyasselmeci) [inline dockerhub.py]: `except: pass`? really?
All that will cause is for 295 to blow up with a NameError on `values`.

Comment (djw8605) [reply dockerhub.py]: Probably used for debugging.  Removing.

**Pull request merged**